docs/book/programs/xdp.md

```diff
@@ -1,3 +1,482 @@
 # XDP
 
-This page is a work in progress, please feel free to open a Pull Request!
+!!! example "Source Code"
+
+    Full code for the example in this chapter is available [here](https://github.com/aya-rs/book/tree/main/examples/xdp-drop)
+
+
+## What is XBP in eBPF?
+
+XBP (eXpress Data Path) is a type of eBPF program that attaches to the network interface. It allows for the processing of network packets as soon as they are received from the network driver, even before they enter the networking stack. XDP enables efficient packet filtering, manipulation and redirection at the earliest possible stage, resulting in low-latency and high-throughput.
```

Comment:
Can we stick to 80-100 character limit per line?

Reply:
> Can we stick to 80-100 character limit per line?

Not sure what you mean per line, you want to break to a newline after max 100 chars?
On my side I don't see the lines go further than > 100 chars, it's always around 90 chars max per line.

Reply:
Yes, I want to break to a new line after max 100 characters.

This exact sentence which I commented exceeds this limit and it's quite hard to read in some editors. The whole line 10 has 317 characters:

```
~> python3                                                                                                       04/30/24 13:30:49 PM
Python 3.11.5 (main, Aug 24 2023, 15:09:45) [Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> len("XDP (eXpress Data Path) is a type of eBPF program that attaches to the network interface. It enables filtering, manipulation and refirection of network packets as soon as they are received from the network driver, even before they enter the Linux kernel networking stack, resulting in low latency and high throughput.")
317
```

We should probably introduce some markdown linter to catch stuff like that, but for now can we just split the obviously too large lines?

Reply:
> Yes, I want to break to a new line after max 100 characters.
> 
> This exact sentence which I commented exceeds this limit and it's quite hard to read in some editors. The whole line 10 has 317 characters:
> 
> ```
> ~> python3                                                                                                       04/30/24 13:30:49 PM
> Python 3.11.5 (main, Aug 24 2023, 15:09:45) [Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
> Type "help", "copyright", "credits" or "license" for more information.
> >>> len("XDP (eXpress Data Path) is a type of eBPF program that attaches to the network interface. It enables filtering, manipulation and refirection of network packets as soon as they are received from the network driver, even before they enter the Linux kernel networking stack, resulting in low latency and high throughput.")
> 317
> ```
> 
> We should probably introduce some markdown linter to catch stuff like that, but for now can we just split the obviously too large lines?

Sure, makes sense, the only thing I'm worried about is that it may be less nice to read on bigger screens or on the site directly

Reply:
For example, on a bigger screen/ on the site, it would look like:

```
XBP (eXpress Data Path) is a type of eBPF program that attaches to the network interface.
It allows for the processing of network packets as soon as they are received from the network driver, even before they enter the networking stack.
XDP enables efficient packet filtering, manipulation and redirection at the earliest possible stage, resulting in low-latency and high-throughput.
```

Maybe it's just me but I find it harder to read with all the line breaks here

Reply:
> For example, on a bigger screen/on the site, it would look like:
> 
> ![image](https://private-user-images.githubusercontent.com/106669955/326723715-052f213c-6677-45cc-aeaf-4e6cf1ae79d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwODY4MDAsIm5iZiI6MTcxNTA4NjUwMCwicGF0aCI6Ii8xMDY2Njk5NTUvMzI2NzIzNzE1LTA1MmYyMTNjLTY2NzctNDVjYy1hZWFmLTRlNmNmMWFlNzlkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxMjU1MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MzE5ZDZlYWM5OWNlOWE1ODcyYzRmZGI1OThlMTA0MzViOWYxYzg5NzI3MzI5YjdkN2ZmMDcwM2I2YzEyZDkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eMUSa4WdxQWoAXrobL9xchjHG4fvYyf5Gy6dwPRkeEI)
> 
> Maybe it's just me but I find it harder to read with all the line breaks here

Any news? I don't mind adding the line breaks, just thought it would make sense to reconsider it.

Reply:
> OK, I had no idea that breaking the lines even with a single `\n` makes it being rendered on the website. Let me test myself. If that's really the case, let's leave the long lines.

@vadorovsky my bad, I tried with `<br>` for line breaks (for some reason I thought you'd also like to render those on the website), going to try with singe `\n`, it shouldn't render on the website.

Reply:
@vadorovsky indeed it doesn't change anything on the website when adding single `\n`, I have added some line breaks for bigger sentences, let me know if I should change anything else :+1: